lib/solar.js

```diff
@@ -147,6 +150,26 @@ function sin_elevation(ms, lat, lon) {
   return sin_lat * sin_dec + cos_lat * cos_dec * Math.cos(hour_angle);
 }
 
+// NREL spa.c#topocentric_azimuth_angle_astro
+// https://midcdmz.nrel.gov/sampa/
+// https://midcdmz.nrel.gov/spa/spa.h
+function azimuth(ms, lat, lon) {
+  const t = julian.to(ms);
+  const sin_lat = Math.sin(lat * RAD);
+  const cos_lat = Math.cos(lat * RAD);
+  const hour_angle = sidereal.local(t, lon * RAD) - _right_ascension(t);
+  const sin_hour_angle = Math.sin(hour_angle);
+  const cos_hour_angle = Math.cos(hour_angle);
+  const dec = _declination(t);
+  const tan_dec = Math.tan(dec);
+
+  return Math.PI + Math.atan2(
+    sin_hour_angle,
+    cos_hour_angle * sin_lat - tan_dec * cos_lat
+  );
+}
+
+
 // Solar position classes. Helpful if you need to cache results!
 class Location {
```

Comment:
I would love it if the `Location` class included azimuth. That's probably work for another day, though.